targeteditor/src/main/scala/explore/targeteditor/TargetQueries.scala

```diff
@@ -73,6 +76,14 @@ object TargetQueries {
   val targetRA: Lens[TargetResult, RightAscension] =
     TargetResult.tracking ^|-> properMotionRA
 
+  /**
+   * Lens to the name of a sidereal target
+   */
+  val unsafeTargetName: Lens[TargetResult, NonEmptyString] =
```

Comment:
This is unsafe lens to be used in undo, It will be protected of misuse by the ui while the api layer is updated to support refined scalars.